src/cplscheme/CouplingData.hpp

```diff
@@ -36,16 +50,15 @@ struct CouplingData {
   }
 
   CouplingData(
-      Eigen::VectorXd *values,
-      mesh::PtrMesh    mesh,
-      bool             requiresInitialization,
-      int              dimension)
-      : values(values),
+      mesh::PtrData data,
+      mesh::PtrMesh mesh,
+      bool          requiresInitialization)
+      : data(data),
         mesh(mesh),
-        requiresInitialization(requiresInitialization),
-        dimension(dimension)
+        requiresInitialization(requiresInitialization)
   {
-    PRECICE_ASSERT(values != NULL);
+    PRECICE_ASSERT(data != nullptr);
+    PRECICE_ASSERT(mesh != nullptr);
```

Comment:
This introduces an invariance to this `struct`, which cannot be maintained due to the public member access.
From a design standpoint, this should now be a `class`, which protects `data` and `mesh`. Let's postpone this change to another time though.

Reply:
Meaning: Don't change it to a `class` now, but keep it on the radar? I would suggest to at least add a `@todo` in the code. See https://github.com/precice/precice/pull/865/files/3fd1911ecfe116124b84ddbc2f876d1b12fb99d1#r495835946.

Reply:
Resolved by ca62d12